components/layout/fragment.rs

```diff
@@ -1051,7 +1051,8 @@ impl Fragment {
                     // Note: We can not precompute the ratio and store it as a float, because
                     // doing so may result one pixel difference in calculation for certain
                     // images, thus make some tests fail.
-                    inline_size * intrinsic_block_size.0 / intrinsic_inline_size.0
```

Comment:
Ouch, right. Maybe we should add a helper method to app_units for this (`muldiv`?). But of course it doesn't need to block this.